### PR TITLE
OAK-9722 Allow replacing default Azure SAS token generator

### PR DESCRIPTION
**Pull request by jelmini**

### CODE REVIEW

**Comment by jelmini** [comment]
@smiroslav @mattvryan Could you please review this PR? Thanks.

**Review by smiroslav — APPROVED** [review]

File: oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStore.java [hunk 1]
```diff
@@ -41,17 +42,24 @@
...
+    private SasTokenGenerator sasTokenGenerator;
+
     private AzureBlobStoreBackend azureBlobStoreBackend;
 
     @Override
     protected AbstractSharedBackend createBackend() {
         azureBlobStoreBackend = new AzureBlobStoreBackend();
         if (null != properties) {
             azureBlobStoreBackend.setProperties(properties);
+            azureBlobStoreBackend.setSasTokenGenerator(sasTokenGenerator);
```
Comment (mreutegg) [inline oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStore.java:54]: This doesn't look right to me. Why set the `sasTokenGenerator` only when there are `properties`?

File: oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/sas/SasTokenGenerator.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.blob.SharedAccessBlobHeaders;
+import com.microsoft.azure.storage.blob.SharedAccessBlobPolicy;
+import java.security.InvalidKeyException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface SasTokenGenerator {
```
Comment (mreutegg) [inline oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/sas/SasTokenGenerator.java:29]: I think this interface should have some basic documentation and the new `sas` package should have an explicit export version in a `package-info.java` file. I understand the interface is intended to be implemented by 3rd parties and therefore must have the export version managed explicitly.

**Review by mreutegg — COMMENTED** [review]
This PR also contains changes unrelated to this issue under oak-lucene.

**Comment by jelmini** [comment]
> This PR also contains changes unrelated to this issue under oak-lucene.

Hmm, weird. I have pushed a single commit with changes only to oak-blob-cloud-azure, but now this PR thinks other unrelated commits were added to it... I don't know how that could happen. 
In case we continue with this PR, I will rebase the relevant commit on top of trunk to get rid of those spurious commits.

File: oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/sas/SasTokenGenerator.java (hunk 2 above)
Comment (jelmini) [reply oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/sas/SasTokenGenerator.java:29]: Yes, during tests I realised explicit export version should be added.
This PR should have been a draft PR, as the goal was to receive early feedback.

File: oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStore.java (hunk 1 above)
Comment (jelmini) [reply oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStore.java:54]: Right. It should be outside the `if` clause.

**Pull request closed without merging**